crates/warpgate/src/loader.rs

```diff
@@ -150,6 +156,102 @@ impl PluginLoader {
         .await
     }
 
+    /// Load a plugin from an OCI registry.
+    #[instrument(skip(self))]
+    pub async fn load_plugin_from_oci<I: AsRef<Id> + Debug>(
+        &self,
+        id: I,
+        locator: &warpgate_api::OciLocator,
+        registries: Vec<OciRegistry>,
+    ) -> Result<PathBuf, WarpgateLoaderError> {
+        let id = id.as_ref();
+        let client = Client::new(ClientConfig::default());
+
+        let location = if locator.image.as_str().starts_with("oci://") {
```

Comment:
Image should never contain `oci://`, that should be stripped during locator parsing.